scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Check for decrease in coverage from 100% of frontend files."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+import re
+import shutil
+import subprocess
+import sys
+
+import python_utils
+
+from . import fully_covered_files
+
+LCOV_FILE_PATH = os.path.join(os.pardir, 'karma_coverage_reports', 'lcov.info')
+PR_LCOV_FILE_PATH = os.path.join(os.curdir, 'tmp', 'pr-lcov.info')
+FULLY_COVERED_TESTS = fully_covered_files.fully_covered_files_array
+
+
+def run_frontend_tests_script():
+    """Run the frontend tests script using subprocess."""
+    subprocess.check_call([
+        'python', '-m', 'scripts.run_frontend_tests'])
+
+
+def create_tmp_folder():
+    """Creates a temporary folder."""
+    if not os.path.exists('./tmp'):
+        os.mkdir('./tmp')
+
+
+def delete_tmp_folder():
+    """Delete the temporary folder."""
+    if os.path.exists('./tmp'):
+        shutil.rmtree('./tmp')
+
+
+def filter_lines(line):
+    """Check if the line has file path (SF) or total lines (LF) or covered
+    lines (LH) of the test.
+
+    Args:
+        line: str. A line from lcov file.
+
+    Returns:
+        Boolean. If the line has the file path or total lines or covered lines
+        of the test.
+    """
+    return ('SF' in line or 'LH' in line or 'LF' in line)
+
+
+def get_test_file_name(test_path):
+    """Get the file name from the absolute path.
+
+    Args:
+        test_path: string. The file's absolute path.
+
+    Returns:
+        String. The file path.
+    """
+    if not test_path:
+        sys.stderr.write(
+            'The test path is empty or null.'
+            'It\'s not possible to diff the test coverage correctly.')
+        sys.exit(1)
+
+    file_name = os.path.split(test_path)[1]
+    return file_name
+
+
+def get_lcov_file_tests(file_path):
```

Comment:
You're not really getting "tests", here -- instead, you are getting lcov lines or lcov stanzas or something like that.

Please name this function more appropriately -- e.g. get_relevant_lines_from_lcov_file() or something like that.

Throughout this function, please don't use "tests" if you're not working with objects that represent tests; instead use "lcov stanzas" or similar. Consider also describing in this docstring how the lcov output is formatted, for the benefit of the developer who is reading the code.

Finally, if you are returning a list of complicated things, consider creating a simple class-based domain object for those instead of putting stuff in lists of lists.

Reply:
I've never used class in Python. Do you think is it the better approach for this function? Is it ok to have a class declared in the middle of the file? Should I refactor all the script to have only classes instead functions?

Reply:
We discussed this in our meeting, and she gets the idea.. She said she will update the PR.